### PR TITLE
Use pre-built images

### PR DESCRIPTION
**Pull request by patrys**
This uses Docker images for Core and Dashboard. It also drop the Storefront. If we want to provide good dockerized experience for people working on the storefront, we can add a [dev container](https://containers.dev) config to its repository.

### CODE REVIEW

File: .github/workflows/test-platfomr.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: test-upgrade-submodules
+name: test-platform
```
Comment (lkostrowski) [inline .github/workflows/test-platfomr.yml:1]: typo in the filename

**Review by lkostrowski — CHANGES_REQUESTED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Review by lkostrowski — APPROVED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Review by peelar — APPROVED** [review]

**Comment by mhcassiem** [comment]
hi, what was the motivation behind removing storefront from platform?
How would you recommend allowing storefront to hit saleor-core?

**Comment by andrzejewsky** [comment]
Hey @mhcassiem - the motivation here is to deliver an environment that you can use to build something on top of Saleor, for example, a storefront, checkout, an external app, and so on. By using the platform you should be able to easily bootstrap Saleor (core API + its dashboard) without digging into details. 

Storefront or the other external projects that are talking to Saleor, you can quickly bootstrap by using their own development mode (simply using a package manager) and configuring where the API is (in the case of saleor platform, dockerized version is on localhost:8000).

**Comment by patrys** [comment]
As I mentioned in the top comment, we want to ensure each project offers an optimized development experience. Platform was a poor fit for that. It ended up being neither suitable for development nor production use. Most people don't need to edit the core or the dashboard, so we want them to be deployable quickly (and using official images) and in tandem with other repositories offering dev container setup optimized for working on storefronts/checkout/payments/etc.

Including everything in the platform does not scale, we have dozens of apps and other satellite services using Saleor, and you don't want them all running on your laptop so that you can edit the storefront.

**Comment by mhcassiem** [comment]
Thank you. I've noticed build times are much faster now. However I encountered an error with react-storefront. 
I was getting ECONNREFUSED for the graphql endpoint.

This was due to the nginx proxying. However, I was running storefront from its own compose file and because of nginx could not proxy to api:8000. I've found a fix in the meantime by connecting the storefront services to the same docker network as saleor-core.

I assume I should probably open an issue in react-storefront for this?

**Comment by patrys** [comment]
@mhcassiem Yes, please open a ticket in the storefront repository. We'll get it fixed.

**Pull request merged**